src/domain/schemas/lesson-schemas.js

```diff
@@ -1,6 +1,10 @@
 import joi from 'joi';
 import { idOrKeySchema, slugSchema, sectionDBSchema } from './shared-schemas.js';
 
+const scheduleSchema = joi.object({
+  startsOn: joi.string().required()
```

Comment:
I think this needs to be instanceOf(Date) so you don't get the warining.

Reply:
Actually in both cases it's `string` here, as the data comes as a JSON from the POST req body.
However we should actually save `Date` in the DB, so I've split the schemas and casted to date upon persisting, so thanks for leading me to notice this.